### ISSUE TITLE
Is there a way to make sure dependants are automatically queued at the front of the queue?

### ISSUE THREAD

**Issue by jtfidje** [rq/rq#1690]
Currently I have a custom Queue class that implements this feature by overriding the `enqueue_dependents` method, but after upgrading to 1.11 stuff breaks as a consequence of my overrides.

Is there a way to achieve this functionality without custom code?

**Comment by selwin** [comment]
I'm happy to accept a PR for this. The newly introduced `Dependency` class gives us a good way of controlling how dependencies are handled. https://github.com/rq/rq/blob/5b95725dc4410634a658bda9aa59188eaeaf87ea/rq/job.py#L41

The API should be as follows:
```python
queue.enqueue(foo, depends_on=Dependency(bar, enqueue_at_front=True))
```

Mind opening a PR for this?

**Comment by jtfidje** [comment]
Sure!

**Comment by lowercase00** [comment]
Implemented with #1696. Closing.
